addons/sale_crm/models/sale_order.py

```diff
@@ -13,3 +13,9 @@ class SaleOrder(models.Model):
 
     def action_confirm(self):
         return super(SaleOrder, self.with_context({k:v for k,v in self._context.items() if k != 'default_tag_ids'})).action_confirm()
+
+    def write(self, vals):
+        res = super().write(vals)
+        if vals.get('state') in ['sent', 'sale', 'cancel'] and self.opportunity_id:
+            self.opportunity_id.with_context(tracking_disable=False)._update_revenues_from_so()
+        return res
```

Comment:
I still believe we only wanted to update when reaching sent and sale states only. If we want to check the value has changed, maybe we could do this:

```python
     def write(self, vals):
        state_per_order = {}
        if vals.get('state') in ['sent', 'sale']:
            state_per_order = {
                order.id: order.state for order in self
            }
        res = super().write(vals)
        if state_per_order:
            new_state_orders = self.filtered(lambda order: order.state != state_per_order.get(order.id))
            if new_state_orders.opportunity_id:
                new_state_orders.opportunity_id.with_context(tracking_disable=False)._update_revenues_from_so()
        return res
```